### PR TITLE
Refactor nodePool controller

### PR DESCRIPTION
**Pull request by enxebre**
Refactor the nodePool controller to match common patterns:
- Manifest generation and reconciliation for managed resources.
- Drop explicit requeueing in favour of watching managed resources.

### CODE REVIEW

**Comment by enxebre** [comment]
/hold
Need to test a few things.

**Review by csrwng — COMMENTED** [review]
A couple of comments, and in general, I think it would make sense for you to set owner references on each of the resources the controller creates.

**Comment by enxebre** [comment]
>A couple of comments, and in general, I think it would make sense for you to set owner references on each of the resources the controller creates.

unfortunately owner references can't cross namespaces.

File: hypershift-operator/controllers/nodepool/manifests.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+			},
+			MaxUnhealthy: &maxUnhealthy,
+			NodeStartupTimeout: &metav1.Duration{
+				Duration: 10 * time.Minute,
+			},
+		},
+	}
+}
+
+func AWSMachineTemplate(infraName, ami string, nodePool *hyperv1.NodePool) *capiaws.AWSMachineTemplate {
```
[the remark this answers is not in the document]
Comment (enxebre) [reply hypershift-operator/controllers/nodepool/manifests.go:43]: this one I'll address it in a follow up as part of this TODO https://github.com/openshift/hypershift/pull/142/files#diff-c666a2a9dc48b0d3de33cc8fa3a054f853671f0428f9d56b263750dfc6d26e00R228-R232

**Comment by csrwng** [comment]
/test aws-e2e
/lgtm
/hold 
Feel free to remove the hold if e2e passes

**Pull request merged**